### PR TITLE
RFC 1: Trust The Closing Wallet to Pick Target Wallets

### PR DESCRIPTION
**Pull request by beaushinkle**
The main idea is that if the wallet were malicious, it would probably
just steal the funds outright. The only real reason for the malicious
wallet to transfer the funds to a wrong-but-valid wallet would be one
that gives the malicious wallet some sort of advantage, like *another*
wallet it controls.

For instance, if the malicious operator set controlled *two* wallets, and
wanted to drain the funds from both of them, it could intentionally make
of them fail a heartbeat, send the funds to the other one, and then
steal the funds from just the other wallet and get slashed once instead
of twice.

In this case, the loss of opportunity of slashed tokens relative to the
loss of bitcoin is so small, and the chance of one operator group being
able to control two wallets simultaneously is so small that I don't
think it's worth complicating the wallet closure protocol for v2.0
(maybe for v2.1 or v2.2). Simplify, simplify, simplify.

tagging @pdyraga and @michalinacienciala for review

### CODE REVIEW

**Review by michalinacienciala — COMMENTED** [review]
I left couple of comments. Some are not strictly related to this PR, they don't need to be addressed here.

Two other things to keep in mind:
1. It's currently being researched if we can add a check to the  `intention-to-move` transaction that validates if the number of chosen target wallets in correct (`min(#_of_live_wallets, balance/wallet_transfer_max)`). If this check gets added, we could mention that in this sentence:
> After this transaction is complete, <<public-knowledge,the public>> is able to submit a SPV proof to let ethereum know that the funds were transferred nonfraudulently (they match the `intention-to-move` wallets, the fee isn't too high, and the funds are split evently)
2. Not strictly related to this PR, but to the 'Closing A Wallet'  section in general, may require separate PR: we decided to differentiate between CLOSING and CLOSED state  and add another parameter, `closing_period` (name can be changed) which defines after what time since wallet went to the CLOSING state can it change its state to CLOSED. The parameter must be equal to at least `locktime` + time for requesting and processing a refund in case wallet decided to steal + time for defeat fraud challenge. Also, a new `public-knowledge` transaction will be possible - one informing that the wallet passed the `closing_period`.
We should mention it somewhere.

File: docs/rfc/rfc-1.adoc [hunk 1]
```diff
@@ -665,25 +665,19 @@ remaining, or exceeds the `wallet_max_age`, the operators begin the process of
 closing it. If there are still funds remaining, and those funds are at least
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply docs/rfc/rfc-1.adoc:665]: good catch!

File: docs/rfc/rfc-1.adoc [hunk 2]
```diff
@@ -665,25 +665,19 @@ remaining, or exceeds the `wallet_max_age`, the operators begin the process of
...
-before `move_funds_start_timestamp` that _wasn't_ chosen _and_ should have
-been. See <<wallet-closure-transaction,the dedicated section>> for how wallets
-are chosen.
-
-If the challenge period passes, that operator proposes a BTC fee (as in
-<<sweeping,sweeping>>) and the operators transfer the funds evenly between the
-wallets proposed in the `intention-to-move` transaction. They construct a P2PKH
-transaction moving the wallet's main UTXO to each of those wallets. For more
-details on exactly how a wallet is chosen and how the funds are split up see
+the operators intend to move the funds to. We trust the wallet to do the
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply docs/rfc/rfc-1.adoc]: [ce98183](https://github.com/keep-network/tbtc-v2/pull/190/commits/ce98183ef73062e9b852c664167e9a8b40c9b2d1)

File: docs/rfc/rfc-1.adoc [hunk 3]
```diff
@@ -665,25 +665,19 @@ remaining, or exceeds the `wallet_max_age`, the operators begin the process of
 closing it. If there are still funds remaining, and those funds are at least
 `wallet_dust_leftover`, then the <<operator-only,first operator>> posts a
 <<operator-only,reimbursable>> transaction to the ethereum chain declaring an
-intention to close the wallet. This transaction includes a timestamp hereby
-referred to as `move_funds_start_timestamp`. If the balance is less than
+intention to close the wallet. If the balance is less than
 `wallet_dust_leftover`, we simply close the wallet and abandon the funds. We
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply docs/rfc/rfc-1.adoc]: Included in [2b92a38](https://github.com/keep-network/tbtc-v2/pull/190/commits/2b92a386a43b22936700fcc1fb3482a1e45761ae)

File: docs/rfc/rfc-1.adoc [hunk 4]
```diff
@@ -665,25 +665,19 @@ remaining, or exceeds the `wallet_max_age`, the operators begin the process of
 closing it. If there are still funds remaining, and those funds are at least
 `wallet_dust_leftover`, then the <<operator-only,first operator>> posts a
 <<operator-only,reimbursable>> transaction to the ethereum chain declaring an
-intention to close the wallet. This transaction includes a timestamp hereby
-referred to as `move_funds_start_timestamp`. If the balance is less than
+intention to close the wallet. If the balance is less than
```
[the remark this answers is not in the document]
Comment (beaushinkle) [reply docs/rfc/rfc-1.adoc]: [2b92a38](https://github.com/keep-network/tbtc-v2/pull/190/commits/2b92a386a43b22936700fcc1fb3482a1e45761ae)

File: docs/rfc/rfc-1.adoc [hunk 5]
```diff
@@ -661,29 +661,31 @@ Governable Parameters:
...
-`wallet_dust_leftover`, then the <<operator-only,first operator>> posts a
-<<operator-only,reimbursable>> transaction to the ethereum chain declaring an
-intention to close the wallet. This transaction includes a timestamp hereby
-referred to as `move_funds_start_timestamp`. If the balance is less than
-`wallet_dust_leftover`, we simply close the wallet and abandon the funds. We
-can maintain the peg via a <<donate,donation>>.
+remaining, or exceeds the `wallet_max_age`, the a few different parties can
+begin the process of closing it.
+
+If the wallet failed a heartbeat, then the then the <<operator-only,first
```
Comment (michalinacienciala) [inline docs/rfc/rfc-1.adoc:667]:
```suggestion
If the wallet failed a heartbeat, then the <<operator-only,first
```

File: docs/rfc/rfc-1.adoc [hunk 6]
```diff
@@ -661,29 +661,31 @@ Governable Parameters:
...
 times, ends a <<redemption,redemption>> with less than `wallet_min_closure_btc`
-remaining, or exceeds the `wallet_max_age`, the operators begin the process of
-closing it. If there are still funds remaining, and those funds are at least
-`wallet_dust_leftover`, then the <<operator-only,first operator>> posts a
-<<operator-only,reimbursable>> transaction to the ethereum chain declaring an
-intention to close the wallet. This transaction includes a timestamp hereby
-referred to as `move_funds_start_timestamp`. If the balance is less than
-`wallet_dust_leftover`, we simply close the wallet and abandon the funds. We
-can maintain the peg via a <<donate,donation>>.
+remaining, or exceeds the `wallet_max_age`, the a few different parties can
```
Comment (michalinacienciala) [inline docs/rfc/rfc-1.adoc:664]:
```suggestion
remaining, or exceeds the `wallet_max_age`, then a few different parties can
```

**Review by michalinacienciala — COMMENTED** [review]
I noticed two small typos. Other than that, LGTM.

File: docs/rfc/rfc-1.adoc (hunk 6 above)
Comment (beaushinkle) [reply docs/rfc/rfc-1.adoc]: [e8dcde7](https://github.com/keep-network/tbtc-v2/pull/190/commits/e8dcde78c75759c62b774ee166d980b8420589e9)

**Review by michalinacienciala — APPROVED** [review]
LGTM. @pdyraga, do you wan't to review this one, or should I merge?

**Review by pdyraga — APPROVED** [review]

**Pull request merged**